pkg/vpa/service.go

```diff
@@ -149,6 +149,44 @@ func (c *Service) CreateTortoiseUpdaterVPA(ctx context.Context, tortoise *autosc
 	return vpa, tortoise, nil
 }
 
+func (c *Service) UpdateVPACRP(ctx context.Context, tortoise *autoscalingv1beta3.Tortoise) (*v1.VerticalPodAutoscaler, error) {
+	retVPA := &v1.VerticalPodAutoscaler{}
+
+	// we only want to record metric once in every reconcile loop.
+	//metricsRecorded := false
+	updateFn := func() error {
+		vpa, err := c.GetTortoiseUpdaterVPA(ctx, tortoise)
+		if err != nil {
+			return fmt.Errorf("get tortoise VPA: %w", err)
+		}
+		crp := make([]v1.ContainerResourcePolicy, 0, len(tortoise.Spec.ResourcePolicy))
+		for _, c := range tortoise.Spec.ResourcePolicy {
+			crp = append(crp, v1.ContainerResourcePolicy{
+				ContainerName: c.ContainerName,
+				MinAllowed:    c.MinAllocatedResources,
+			})
+		}
+		vpa.Spec.ResourcePolicy.ContainerPolicies = crp
+		retVPA = vpa
+		if tortoise.Spec.UpdateMode == autoscalingv1beta3.UpdateModeOff {
+			// don't update status if update mode is off. (= dryrun)
+			return nil
+		}
```

Comment:
We can update VPA even if tortoise is in Off mode because VPA is fully managed by tortoises and not something created by users. (= users don't even know VPA is working under the hood.)